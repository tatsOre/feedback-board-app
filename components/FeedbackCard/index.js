import Link from 'next/link'
import UpvoteButton from '../Buttons/Upvote'
import { getCommentsLength, toCapitalize } from '../../utils/index'

export default function FeedbackCard({ feedback }) {
  const { title, description, category, upvotes, comments, slug } =
    feedback

  const commentsLength = getCommentsLength(comments)
  const categoryText = ['ux', 'ui'].includes(category)
    ? category.toUpperCase()
    : toCapitalize(category)

  return (
    <>
      <h3 className="text-sm lg:text-lg mb-2">
        <Link href={`/feedback/detail/${slug}`} passHref>
          <a className="text-indigo-800 hover:text-blue-900">{title}</a>
        </Link>
      </h3>
      <p className="text-indigo-500">{description}</p>
      <p className="w-min rounded-10 bg-indigo-300 px-4 py-1.5 mt-2 mb-4 md:mt-3 lg:text-small font-semibold text-blue-900">
        {categoryText}
      </p>
      <UpvoteButton upvoted={false} upvotes={upvotes} variant="inline" />
      <p className="comments float-right mt-2 font-bold lg:text-base text-indigo-800">
        {commentsLength}
      </p>
    </>
  )
}
