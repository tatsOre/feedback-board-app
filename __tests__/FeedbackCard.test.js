import { render, screen } from '@testing-library/react'
import FeedbackCard from '../components/FeedbackCard'
import MOCK_FEEDBACK_DATA from '../__mocks__/mock-feedbacks.json'

describe('<FeedbackCard />', () => {
  const {
    data: { suggestion },
  } = MOCK_FEEDBACK_DATA

  it('renders with correct data and UI', () => {
    const { container } = render(<FeedbackCard data={suggestion[0]} />)

    expect(container).toMatchSnapshot()

    expect(screen.queryByRole('link')).not.toBeInTheDocument()

    expect(screen.getByRole('heading', { level: 1 })).toHaveTextContent(
      suggestion[0].title
    )

    expect(screen.getByRole('button', { name: 'upvote' })).toBeInTheDocument()
  })

  it('renders with correct data and UI with link to Feedback Post', () => {
    const { container } = render(<FeedbackCard data={suggestion[1]} link />)

    expect(container).toMatchSnapshot()

    expect(screen.getByRole('link')).toHaveAttribute(
      'href',
      `/feedback/${suggestion[1].slug}/detail`
    )

    expect(screen.getByRole('heading', { level: 3 })).toHaveTextContent(
      suggestion[1].title
    )
  })
})
