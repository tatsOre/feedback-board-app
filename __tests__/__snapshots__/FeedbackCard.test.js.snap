// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeedbackCard /> renders with correct data and UI 1`] = `
<div>
  <article
    class=" feedback-card w-full flex flex-wrap items-center md:items-start bg-white text-[13px] rounded-10 p-6 md:px-8"
  >
    <div
      class="w-full md:w-[80%] md:ml-10"
    >
      <h1
        class="text-sm lg:text-lg mb-2 text-indigo-800"
      >
        Add tags for solutions
      </h1>
      <p
        class="text-indigo-500"
      >
        Easier to search for solutions based on a specific stack.
      </p>
      <p
        class="w-min rounded-10 bg-indigo-300 px-4 py-1.5 mt-2 mb-4 md:mt-3 lg:text-small font-semibold text-blue-900"
      >
        Enhancement
      </p>
    </div>
    <button
      aria-label="upvote"
      class="bg-indigo-300 text-indigo-800 hover:bg-indigo-400 rounded-10 text-[13px] min-w-[70px] py-3 px-4 leading-3 flex-inline justify-evenly relative z-10"
      type="button"
    >
      <svg
        height="7"
        width="10"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 6l4-4 4 4"
          fill="none"
          fill-rule="evenodd"
          stroke="#4661E6"
          stroke-width="2"
        />
      </svg>
      112
    </button>
    <p
      aria-label="comments"
      class="text-indigo-800 comments ml-auto md:my-auto font-bold lg:text-base"
    >
      2
    </p>
  </article>
</div>
`;

exports[`<FeedbackCard /> renders with correct data and UI with link to Feedback Post 1`] = `
<div>
  <article
    class="group cursor-pointer relative feedback-card w-full flex flex-wrap items-center md:items-start bg-white text-[13px] rounded-10 p-6 md:px-8"
  >
    <a
      aria-label="link to Add a dark theme option"
      href="/feedback/add-a-dark-theme-option/detail"
    >
      <span
        class="full-link"
      />
    </a>
    <div
      class="w-full md:w-[80%] md:ml-10"
    >
      <h3
        class="text-sm lg:text-lg mb-2 text-indigo-800 group-hover:text-blue-900"
      >
        Add a dark theme option
      </h3>
      <p
        class="text-indigo-500"
      >
        It would help people with light sensitivities and who prefer dark mode.
      </p>
      <p
        class="w-min rounded-10 bg-indigo-300 px-4 py-1.5 mt-2 mb-4 md:mt-3 lg:text-small font-semibold text-blue-900"
      >
        Feature
      </p>
    </div>
    <button
      aria-label="upvote"
      class="bg-indigo-300 text-indigo-800 hover:bg-indigo-400 rounded-10 text-[13px] min-w-[70px] py-3 px-4 leading-3 flex-inline justify-evenly relative z-10"
      type="button"
    >
      <svg
        height="7"
        width="10"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 6l4-4 4 4"
          fill="none"
          fill-rule="evenodd"
          stroke="#4661E6"
          stroke-width="2"
        />
      </svg>
      98
    </button>
    <p
      aria-label="comments"
      class="text-indigo-800 comments ml-auto md:my-auto font-bold lg:text-base"
    >
      4
    </p>
  </article>
</div>
`;
